### DISCUSSION TITLE
[SOLVED] Display full article

### DISCUSSION THREAD

**Discussion by senseiski** [martinrotter/rssguard#532] [Q&A] [answered]
Hello, I was reading through the documentation, but I'm not sure if I understand the filters - is there a way to add a script to download the full article content for truncated feeds in the nowebengine app?

**Comment by martinrotter** [comment, marked as answer]
There perfectly is! But there are two features which you mind need to know about:

* [article filters](https://github.com/martinrotter/rssguard/blob/master/resources/docs/Documentation.md#fltr) - they are used to tweak contents of incoming articles from feed and are written in JavaScript
* [post-processing scripts](https://github.com/martinrotter/rssguard/blob/master/resources/docs/Documentation.md#scrap) - they are much more superior to filters and they can totally change structure of feed files, replace whatever they want, perform downloadings of other files, embedding images etc. and can be written in any language you want, particularly good is Python, Bash or Perl.

Both options work in nonwebengine version of RSS Guard. Is there any specific feed/website you have in mind?

**Reply by senseiski** [reply]
Thanks for the prompt reply! What I had in mind was something that could work for most websites, the less technically adept folks that cannot necessarily write our own scripts. The documentation on it seems quite technical so perhaps an example - https://www.dedoimedo.com/rss_feed.xml this feed has only truncated posts - how can I add a script to extract the full article and can it be displayed within RSSGuard nowebengine instead of the few sentences provided in the rss feed? (PS: not quite related but is there a way to get images shown in the nowebengine build?)

**Reply by martinrotter** [reply]
As for images, no at this point, you can click link and download the image, but I might add the feature.

As for your problem. If the website is not loaded via JavaScript but directly then single script would basically work for all RSS feeds etc.

I could try to write one simple if you wish.

**Reply by senseiski** [reply]
That would be great if you can! thanks so much for your help and patience!

**Reply by martinrotter** [reply]
OK, the script actually already exists.

https://github.com/martinrotter/rssguard/blob/master/resources/scripts/scrapers/scrape-as-rss2.py

It should do exactly what you want, it replaces original RSS articles with articles downloaded via their URLs.

**Reply by martinrotter** [reply]
OK, tested. This is result:

![image](https://user-images.githubusercontent.com/1255302/141446606-53851eea-1089-41d0-891b-19d68244e00a.png)

So it seems it works. Here is how feed is setup:

![image](https://user-images.githubusercontent.com/1255302/141446692-530abeba-5865-4230-bd28-0716c0684349.png)

Pay attention to "Post-process script" field.

1. Install python - https://www.python.org/ftp/python/3.10.0/python-3.10.0-amd64.exe
2. Install "asyncio" python extension by running "pip3 install asyncio" command in your admin command line.
3. Copy [script](https://raw.githubusercontent.com/martinrotter/rssguard/master/resources/scripts/scrapers/scrape-as-rss2.py) to your PC and save to some file, for example "C:\scraper.py".
4. Setup your feed to use the scraper file, so your post-process script field would look like this: `python#C:\scraper.py#4` (character `#` is used as separator to separate differenet arguments, number `4` means, that articles will be fetched in 4 parallel threads).

**Reply by martinrotter** [reply]
![Vs4kq40LA3](https://user-images.githubusercontent.com/1255302/141447494-f8c22538-d2e8-48c5-9c6d-872998c9c011.gif)

**Reply by senseiski** [reply]
Thanks a lot! It works great with local RSS, but is there a way to make it work with say Feedly? The Feedly feeds don't have the same edit options.

**Reply by martinrotter** [reply]
No, it cannot AFAIK work with Feedly, sadly.
